RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -143,6 +252,34 @@ extension PurchaseHandler {
         }
     }
 
+    @MainActor
+    func performExternalRestoreLogic() async throws -> (info: CustomerInfo, success: Bool) {
+        Logger.debug(Strings.executing_external_restore_logic)
+
+        guard let externalRestoreMethod = self.performRestore else {
+            throw PaywallError.performPurchaseAndRestoreHandlersNotDefined
+        }
+
+        defer {
+            self.restoreInProgress = false
+            self.actionInProgress = false
+        }
+
+        self.restoreInProgress = true
+        self.restoredCustomerInfo = nil
```

Comment:
In original restorePurchase logic, `self.restoredCustomerInfo` was never set after the restore had executed.

Reply:
Apparently it was done in a Button's handler (?), so `self.setRestored(customerInfo)` has now been added.

Reply:
I _think_ this the right thing to do!